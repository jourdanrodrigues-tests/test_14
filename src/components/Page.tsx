import React from 'react';

import Typography from '@/components/Typography.tsx';
import mergeCls from '@/mergeCls.ts';

type Props = { title: string; children: React.ReactNode; className?: string };

export default function Page({ title, children, className }: Props) {
  return (
    <div
      className={mergeCls(
        'h-full w-full flex flex-col items-center justify-center',
        className
      )}
    >
      <Typography variant="h1">{title}</Typography>
      {children}
    </div>
  );
}
