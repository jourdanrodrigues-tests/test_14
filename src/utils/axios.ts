import { AxiosError } from 'axios';

const NETWORK_CODES = new Set(['ERR_NETWORK', 'ECONNABORTED']);

export function isOffline(e: unknown) {
  return !!(e instanceof AxiosError && e.code && NETWORK_CODES.has(e.code));
}

export function isServerError(e: unknown): boolean {
  return (
    e instanceof AxiosError && Math.floor((e.response?.status || 0) / 100) === 5
  );
}
