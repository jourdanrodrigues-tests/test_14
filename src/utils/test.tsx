import React from 'react';

import type { UseQueryResult } from '@tanstack/react-query';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render as rtlRender, waitFor } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { expect } from 'bun:test';
import pretty from 'pretty';

type TQueryHooks<
  TData = unknown,
  TError = unknown,
  QueryResult = UseQueryResult<TData, TError>,
> = Array<{
  useHook(): QueryResult;
  validate(query: QueryResult): boolean;
}>;

type RenderOptions = {
  queryHooks?: TQueryHooks;
};

export const testQueryClient = new QueryClient();
const queryClientWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

export async function render(
  node: React.ReactElement,
  options?: RenderOptions
) {
  const hookPromises = (options?.queryHooks || []).map(
    ({ useHook, validate }) => {
      const { result } = renderHook(useHook, { wrapper: queryClientWrapper });
      return waitFor(() => validate(result.current));
    }
  );
  if (hookPromises.length > 0) {
    await Promise.all(hookPromises);
  }

  const result = rtlRender(node, {
    wrapper: ({ children }) => wrapNode(children),
  });
  return result.container;
}

function wrapNode(node: React.ReactNode): React.ReactNode {
  return queryClientWrapper({ children: node });
}

export function expectSnapshot(element: HTMLElement | null) {
  // TODO: Use @testing-library/jest-dom instead of @happy-dom
  //  The JestDOM does some HTML pretty-printing that is better for diffs
  return expect(element && pretty(element.innerHTML));
}
