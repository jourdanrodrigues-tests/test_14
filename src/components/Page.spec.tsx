import { describe, it } from 'bun:test';

import Page from '@/components/Page.tsx';
import { expectSnapshot, render } from '@/utils/test';

describe('Page Component', () => {
  it('should match the snapshot', async () => {
    const result = await render(
      <Page className="custom-page-class" title="Page title">
        <span>Page content</span>
      </Page>
    );

    expectSnapshot(result.container).toMatchSnapshot();
  });
});
