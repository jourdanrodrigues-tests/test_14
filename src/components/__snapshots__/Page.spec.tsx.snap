// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component should match the snapshot 1`] = `
"<div class="max-h-screen w-screen flex flex-col items-center justify-center custom-page-class">
  <h1 class="block antialiased tracking-normal font-sans text-5xl font-semibold leading-tight text-inherit">Page title</h1><span>Page content</span>
</div>"
`;
